gemstone/src/swapper/mayan/swift_provider.rs

```diff
@@ -0,0 +1,559 @@
+                msg: "Missing evmSwapRouterAddress".to_string(),
+            })?;
+            let evm_swap_router_calldata = mayan_quote.evm_swap_router_calldata.clone().ok_or_else(|| SwapperError::ComputeQuoteError {
+                msg: "Missing evmSwapRouterCalldata".to_string(),
+            })?;
+            let min_middle_amount = mayan_quote.min_middle_amount.ok_or_else(|| SwapperError::ComputeQuoteError {
+                msg: "Missing minMiddleAmount".to_string(),
+            })?;
+
+            let token_in = mayan_quote.from_token.contract.clone();
+            let formatted_min_middle_amount = self
+                .get_amount_of_fractional_amount(min_middle_amount, mayan_quote.swift_input_decimals)
+                .map_err(|e| SwapperError::ComputeQuoteError { msg: e.to_string() })?;
+
+            let amount_in = U256::from_str(quote.from_value.as_str()).map_err(|_| SwapperError::InvalidAmount)?;
+            let swap_data = hex::decode(evm_swap_router_calldata.trim_start_matches("0x")).map_err(|_| SwapperError::ABIError {
+                msg: "Failed to decode evm_swap_router_calldata hex string without prefix 0x ".to_string(),
+            })?;
+            let min_middle_amount = U256::from_str(&formatted_min_middle_amount).map_err(|_| SwapperError::InvalidAmount)?;
+
+            if mayan_quote.from_token.contract == MAYAN_ZERO_ADDRESS {
+                forwarder
+                    .encode_swap_and_forward_eth_call(
+                        amount_in,
+                        evm_swap_router_address.as_str(),
+                        swap_data,
+                        mayan_quote.swift_input_contract.as_str(),
+                        min_middle_amount,
+                        swift_address.as_str(),
+                        swift_call_data,
+                    )
+                    .await
+                    .map_err(|e| SwapperError::ABIError { msg: e.to_string() })?
+            } else {
+                value = "0".to_string();
+                forwarder
+                    .encode_swap_and_forward_erc20_call(
+                        token_in.as_str(),
+                        amount_in,
+                        None,
+                        evm_swap_router_address.as_str(),
+                        swap_data,
+                        mayan_quote.swift_input_contract.as_str(),
+                        min_middle_amount,
+                        swift_address.as_str(),
+                        swift_call_data,
+                    )
+                    .map_err(|e| SwapperError::ABIError { msg: e.to_string() })?
+            }
+        };
+
+        Ok(SwapQuoteData {
+            to: MAYAN_FORWARDER_CONTRACT.to_string(),
+            value: value.clone(),
+            data: forwarder_call_data.encode_hex(),
+        })
+    }
+
+    async fn get_transaction_status(&self, chain: Chain, transaction_hash: &str, provider: Arc<dyn AlienProvider>) -> Result<bool, SwapperError> {
+        todo!();
```

Comment:
is there any endpoint to query this tx status? we can add it later for this method